Struct/array tag patches

wz2b opened the pull request:
This pull request includes two changes:
  - Modify CNSymbolPath to allow brackets and dots in tag paths
  - Modify EtherNetIP to add readTag('tagname', count); for retrieving arrays

This was tested against an Allen Bradley 1769-L24 from the 5370 series of PLCs and confirmed to work, and not interfere with the ability to retrieve single, scalar tags as before.  Further, the EtherNetIP interface was extended (not modified) so none of this should break any existing code.

wz2b commented:
Please build and confirm everything is OK first ... let me know if there were issues.

kasemir commented:
I've added a basic CNSymbolPath.toString() so I can see the path in the debugger and when printing.

wz2b commented:
Thank you - I didn't know enough to figure that out.  I also added two new
unit tests ...


On Thu, Dec 15, 2016 at 3:34 PM, kasemir <notifications@github.com> wrote:

> I've added a basic CNSymbolPath.toString() so I can see the path in the
> debugger and when printing.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/EPICSTools/etherip/pull/11#issuecomment-267436662>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABc2OYBXYfucwcJCfnhxrBQHvlMEutagks5rIaRggaJpZM4LOYWW>
> .
>
